File: 3689. maximum-total-subarray-value-i.py
from typing import List

class Solution:
    def maxTotalValue(self, A: List[int], k: int) -> int:
        gMin = gMax = A[0]

        for n in A:
            gMin = min(gMin, n)
            gMax = max(gMax, n)

        return (gMax - gMin) * k

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1
    A1 = [1,3,2]
    k1 = 2
    output1 = sol.maxTotalValue(A1, k1)
    print(f"Example 1: A={A1}, k={k1}")
    print(f"Output: {output1}")
    print(f"Expected: 4")
    print()
    
    # Example 2
    A2 = [4,2,5,1]
    k2 = 3
    output2 = sol.maxTotalValue(A2, k2)
    print(f"Example 2: A={A2}, k={k2}")
    print(f"Output: {output2}")
    print(f"Expected: 12")

File: 3689. maximum-total-subarray-value-i.java
class Solution {
    public long maxTotalValue(int[] A, int k) {
        int min = A[0], max = A[0];

        for (int n : A) {
            min = Math.min(min, n);
            max = Math.max(max, n);
        }

        return (long) (max - min) * k;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1
        int[] A1 = {1,3,2};
        int k1 = 2;
        long output1 = sol.maxTotalValue(A1, k1);
        System.out.println("Example 1: A=[1,3,2], k=2");
        System.out.println("Output: " + output1);
        System.out.println("Expected: 4");
        System.out.println();
        
        // Example 2
        int[] A2 = {4,2,5,1};
        int k2 = 3;
        long output2 = sol.maxTotalValue(A2, k2);
        System.out.println("Example 2: A=[4,2,5,1], k=3");
        System.out.println("Output: " + output2);
        System.out.println("Expected: 12");
    }
}